### ISSUE TITLE
Geo_shape intersects query regression for small polygons since 6.6

### ISSUE THREAD

**Issue by baptabl** [elastic/elasticsearch#39792]
**Elasticsearch version**:
`Version: 6.6.1, Build: default/tar/1fd8f69/2019-02-13T17:10:04.160291Z, JVM: 11.0.1`

**Plugins installed**:
`None`

**JVM version** (`java -version`):
`openjdk 11.0.1 2018-10-16`
`OpenJDK Runtime Environment 18.9 (build 11.0.1+13)`
`OpenJDK 64-Bit Server VM 18.9 (build 11.0.1+13, mixed mode)`

**OS version**:
`17.7.0 Darwin Kernel Version 17.7.0: Fri Nov  2 20:43:16 PDT 2018; root:xnu-4570.71.17~1/RELEASE_X86_64 x86_64`

**Description of the problem including expected versus actual behavior**:
A geo_shape intersects query to test if a polygon is in another polygon is failing if the polygon used for searching is too small. Same queries are working in 6.5 and not in 6.6.

**Steps to reproduce**:

 1. Index creation
```
PUT test_index
POST test_index/_mapping/_doc
{
  "properties":
    {
      "location": {
        "type": "geo_shape"
      }
    }
}
```

 2. index a shape
```
POST test_index/_doc
{
  "location": {
    "type": "Polygon",
    "coordinates": [
      [
        [
          2.29664146900177,
          48.86463888986946
        ],
        [
          2.2965314984321594,
          48.86463183200937
        ],
        [
          2.2966280579566956,
          48.86446244306811
        ],
        [
          2.296866774559021,
          48.86449596800823
        ],
        [
          2.2968024015426636,
          48.864589484827796
        ],
        [
          2.29664146900177,
          48.86463888986946
        ]
      ]
    ]
  }
}
```

 3. search if a shape is in the indexed shape (it is)
```
# failing small triangle
GET test_index/_search
{
  "query": {
    "geo_shape": {
      "location": {
        "shape": {
          "type": "Polygon",
          "coordinates": [
            [
              [
                2.296668291091919,
                48.864617716286155
              ],
              [
                2.2967085242271423,
                48.86455772441809
              ],
              [
                2.2967755794525146,
                48.86458066249377
              ],
              [
                2.296668291091919,
                48.864617716286155
              ]
            ]
          ]
        },
        "relation": "intersects"
      }
    }
  }
}

# working bigger triangle
GET test_index/_search
{
  "query": {
    "geo_shape": {
      "location": {
        "shape": {
          "type": "Polygon",
          "coordinates": [
            [
              [
                2.296668291091919,
                48.86461948075176
              ],
              [
                2.2967112064361572,
                48.86453302186331
              ],
              [
                2.2967997193336482,
                48.86457536909265
              ],
              [
                2.296668291091919,
                48.86461948075176
              ]
            ]
          ]
        },
        "relation": "intersects"
      }
    }
  }
}
```

In 6.5 both queries are returning the good hit, in 6.6 only the second one is successful.

**Comment by iverase** [comment]
In 6.6, it has been introduced a new indexing strategy based on Lucene BKD tree and some clever breakdown of polygons into triangles. This is the default strategy and the one used if the mapping is declared as you write in this issue.

This error is a known issue in the Lucene implementation and it is fixed in the next release:

https://issues.apache.org/jira/browse/LUCENE-8654

**Comment by iverase** [comment]
This was fixed in Lucene 7.7 and therefore fixed in Elasticsearch 6.7.
